lib/liquid/combinators/general.ex

```diff
@@ -124,23 +124,46 @@ defmodule Liquid.Combinators.General do
   == != > < >= <=
   """
   def comparison_operators do
-    choice([
-      string(@equals),
-      string(@does_not_equal),
-      string(@greater_than),
-      string(@less_than),
-      string(@greater_or_equal),
-      string(@less_or_equal),
-      string("contains")
-    ])
+    empty()
+    |> choice([
+        string(@equals),
+        string(@does_not_equal),
+        string(@greater_than),
+        string(@less_than),
+        string(@greater_or_equal),
+        string(@less_or_equal),
+        string("contains")
+      ])
+    |> traverse({__MODULE__, :to_atom, []})
+  end
+
+  def to_atom(_rest, [h | _], context, _line, _offset) do
+    {h |> String.to_atom() |> List.wrap(), context}
```

Comment:
Please use `String.to_existing_atom`

Reply:
This is not necessary. This occurs only in compilation time, when macros are generating code. Changing this code will cause fails in compilation because no atom with this names exists.